### PR TITLE
Problem (CRO-627) council node pubkey format different from tendermint

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- serialize TendermintValidatorPubKey the same as tendermint.
- Remove ValidatorPubkey in favor of TendermintValidatorPubKey.

### CODE REVIEW

**Comment by yihuang** [comment]
Oh, this one needs to wait for https://github.com/crypto-com/chain/pull/651

File: chain-core/src/state/tendermint.rs [hunk 1]
```diff
@@ -13,20 +13,53 @@ use std::prelude::v1::{String, ToString, Vec};
...
+
+#[cfg(feature = "base64")]
+fn deserialize_ed25519_base64<'de, D>(deserializer: D) -> Result<[u8; PUBLIC_KEY_SIZE], D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let bytes = base64::decode(String::deserialize(deserializer)?.as_bytes())
+        .map_err(|e| D::Error::custom(format!("{}", e)))?;
+    let mut result = [0u8; PUBLIC_KEY_SIZE];
+    result.copy_from_slice(&bytes);
```
Comment (tomtau) [inline chain-core/src/state/tendermint.rs]: `copy_from_slice` will panic if the lengths don't match, so ideally before this, there should be a check on `bytes.len() == PUBLIC_KEY_SIZE`

File: chain-core/src/state/tendermint.rs [hunk 2]
```diff
@@ -37,15 +70,26 @@ impl fmt::Display for TendermintValidatorPubKey {
 }
 
 impl TendermintValidatorPubKey {
+    #[cfg(feature = "base64")]
+    pub fn from_base64(bytes: &[u8]) -> Result<TendermintValidatorPubKey, base64::DecodeError> {
+        let mut result = [0u8; PUBLIC_KEY_SIZE];
+        result.copy_from_slice(&base64::decode(bytes)?);
```
Comment (tomtau) [inline chain-core/src/state/tendermint.rs]: copy_from_slice will panic if the lengths don't match, so ideally before this, there should be a check on bytes.len() == PUBLIC_KEY_SIZE

File: chain-core/src/state/tendermint.rs (hunk 1 above)
Comment (tomtau) [reply chain-core/src/state/tendermint.rs]: also perhaps just deserialise the string and then call `TendermintValidatorPubKey::from_base64` ?

Comment (yihuang) [reply chain-core/src/state/tendermint.rs]: Fixed, also brings in `thiserror` crate, looks like a decent way to handle error type hierarchies.

**Comment by tomtau** [comment]
bors try

**Comment by yihuang** [comment]
Fixed the integration test script error.

**Comment by tomtau** [comment]
bors retry

**Comment by tomtau** [comment]
bors retry

**Comment by tomtau** [comment]
bors retry

**Comment by tomtau** [comment]
bors retry

**Pull request merged**